support/releaseinfo/registryclient/client.go

```
@@ -223,36 +232,111 @@ func ExtractImageFilesToDir(ctx context.Context, imageRef string, pullSecret []b
+		return nil, reference.DockerImageReference{}, fmt.Errorf("failed to create secure transport: %w", err)
 	}
 	insecureRT, err := rest.TransportFor(&rest.Config{TLSClientConfig: rest.TLSClientConfig{Insecure: true}})
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to create insecure transport: %w", err)
+		return nil, reference.DockerImageReference{}, fmt.Errorf("failed to create insecure transport: %w", err)
 	}
 	credStore, err := dockercredentials.NewFromBytes(pullSecret)
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to parse docker credentials: %w", err)
+		return nil, reference.DockerImageReference{}, fmt.Errorf("failed to parse docker credentials: %w", err)
 	}
 	registryContext := registryclient.NewContext(rt, insecureRT).WithCredentials(credStore).
 		WithRequestModifiers(transport.NewHeaderRequestModifier(http.Header{http.CanonicalHeaderKey("User-Agent"): []string{rest.DefaultKubernetesUserAgent()}}))
 
 	ref, err := reference.Parse(imageRef)
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to parse image reference %q: %w", imageRef, err)
+		return nil, reference.DockerImageReference{}, fmt.Errorf("failed to parse image reference %q: %w", imageRef, err)
 	}
 	repo, err := registryContext.Repository(ctx, ref.DockerClientDefaults().RegistryURL(), ref.RepositoryName(), false)
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to create repository client for %s: %w", ref.DockerClientDefaults().RegistryURL(), err)
+		return nil, reference.DockerImageReference{}, fmt.Errorf("failed to create repository client for %s: %w", ref.DockerClientDefaults().RegistryURL(), err)
 	}
-	firstManifest, location, err := manifest.FirstManifest(ctx, ref, repo)
+	return repo, ref, nil
+}
+
+// GetManifest gets the manifest from a docker image
+func GetManifest(ctx context.Context, imageRef string, pullSecret []byte) (distribution.Manifest, error) {
+	repo, ref, err := GetRepoSetup(ctx, imageRef, pullSecret)
```

why is this error being ignored?

---

shouldn't this be

```
	repo, ref, err := GetRepoSetup(ctx, imageRef, pullSecret)
	if err != nil {
		return nil, err
	}

	var srcDigest digest.Digest
	if len(ref.Tag) > 0 {
		desc, err := repo.Tags(ctx).Get(ctx, ref.Tag)
		if err != nil {
			return nil, err
		}
		srcDigest = desc.Digest
	}

	if len(ref.ID) > 0 {
		srcDigest = digest.Digest(ref.ID)
	}

	manifests, err := repo.Manifests(ctx)
	if err != nil {
		return nil, err
	}

	digestsManifest, err := manifests.Get(ctx, srcDigest, manifest.PreferManifestList)
	if err != nil {
		return nil, err
	}

	return digestsManifest, nil
```

---

It shouldn't have been. I updated the code with your suggested code.